docs/python/doxygenlib/cdWriterDocstring.py

```diff
@@ -687,49 +699,89 @@ def __getFullDoc(self, pyname, pyobj, doxy):
         # make the doxy element static if it is tagged as such
         if ATTR_STATIC_METHOD in doxy.doc['tags']:
             doxy.static = 'yes'
-        
+
         lines = self.__getShortDescription(pyname, pyobj, doxy)
         if doxy.isFunction() and type(pyobj) != property:
-            lines += self.__getSignatureDescription(pyname, pyobj, doxy)
-            lines.append('')
+            description = self.__getSignatureDescription(pyname, pyobj, doxy)
+            if description is not None:
+                lines += description
+                lines.append('')
         lines += self.__getDocumentation(pyname, pyobj, doxy)
         lines.append('')
         return lines
 
-    def __getOutputFormat(self, pypath, pyobj, overloads):
-        """Return the line that installs the docstring into the namespace."""
+    @classmethod
+    def __stripBoostSig(cls, doc):
+        def looksLikeBoostSig(l):
+            return bool(l and (l[0].isalnum() or l[0] == "_") and ' -> ' in l)
+
+        lines = doc.strip().splitlines()
+        if len(lines) and looksLikeBoostSig(lines[0]):
+            # boost signature has been prepended.  that means if there is
+            # an existing description provided in the C++ wrapper it will be
+            # indented.  strip the signature and dedent the description.
+            found = False
+            newLines = []
+            for line in lines:
+                if not found and line.startswith('    '):
+                    found = True
+                if found:
+                    newLines.append(line)
+            return textwrap.dedent('\n'.join(newLines))
+        else:
+            # None indicates that the existing __doc__ attr should be left as-is
+            return None
+
+    def __getDocstring(self, pypath, pyobj, overloads):
+        """Return the docstring."""
 
-        # is there an existing python docstring? we don't want to overwrite
-        # this because it may be custom authored in the C++ wrap files.
-        # However, we always override the module doc string for now...
+        docString = ''
+
+        # boost auto-generates function signatures that can be useful in some
+        # contexts (such as generating pyi type stubs) because they more
+        # accurately reflect changes made to the API within the C++ wrap files,
+        # but for our docstrings we strip them out and replace them with our
+        # own function signatures.  After stripping out the boost signatures,
+        # if there is a custom authored docstring in the C++ wrap files we honor
+        # it.
         if hasattr(pyobj, '__doc__') and pyobj.__doc__ is not None:
             doc = pyobj.__doc__.strip()
-            if len(doc) > 0 and not doc.startswith("C++ signature:") \
-               and not overloads[0].isModule():
-                Debug("Docstring exists for %s - skipping" % pypath)
-                return None
+            if len(doc) > 0 and not overloads[0].isModule():
```

Comment:
Might be nice to add the old `we always override the module doc string for now...` comment back here